### PR TITLE
Add `CenteredFaceCoordinates` compute tag

### PR DESCRIPTION
**Pull request by nikwit**
## Proposed changes
Adds a convenient compute tag that computes the grid coordinates of each element face abutting the worldtube

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 1]
```diff
@@ -66,6 +66,15 @@ struct ExcisionSphere : db::SimpleTag {
     return excision_spheres.at(excision_sphere);
   }
 };
+/*!
+ * \brief An optional that holds the grid coordinates of an element face
+ * abutting the worldtube. If the element does not abut the worldtube, it holds
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp]: "..., centered on the worldtube"

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 2]
```diff
@@ -66,6 +66,15 @@ struct ExcisionSphere : db::SimpleTag {
     return excision_spheres.at(excision_sphere);
   }
 };
+/*!
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:76]: (nitpick) add an empty line above while you're making changes

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 3]
```diff
@@ -76,6 +82,23 @@ struct CenteredFaceCoordinates : db::SimpleTag {
   using type = std::optional<tnsr::I<DataVector, Dim, Frame::Grid>>;
 };
 
+template <size_t Dim>
+struct ComputeCenteredFaceCoordinates : CenteredFaceCoordinates<Dim>,
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp]: `CenteredFaceCoordinatesCompute` (that's our convention for compute tags, helps with readability and alphabetically sorted lists)

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.cpp [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+#include "Evolution/DiscontinuousGalerkin/ProjectToBoundary.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace CurvedScalarWave::Worldtube::Tags {
+template <size_t Dim>
+void ComputeCenteredFaceCoordinates<Dim>::function(
+    const gsl::not_null<std::optional<tnsr::I<DataVector, Dim, Frame::Grid>>*>
+        res,
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.cpp]: (nitpick) `result`

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/Test_Tags.cpp [hunk 5]
```diff
@@ -6,12 +6,22 @@
...
+#include <optional>
 
+#include "Domain/Block.hpp"
+#include "Domain/CreateInitialElement.hpp"
 #include "Domain/Creators/DomainCreator.hpp"
 #include "Domain/Creators/Shell.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/InitializeElementFacesGridCoordinates.hpp"
```
Comment (nilsvu) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/Test_Tags.cpp:18]: (optional) If you want to shorten this I suggest you name the file `InitializeElementFaces.hpp`

File: tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/Test_Tags.cpp [hunk 6]
```diff
@@ -31,6 +41,102 @@ void test_excision_sphere_tag() {
...
+      Tags::ComputeCenteredFaceCoordinates<Dim>>("CenteredFaceCoordinates");
+
+  for (const auto& [initial_refinement, quadrature] :
+       cartesian_product(std::array<size_t, 2>{{0, 1}},
+                         make_array(Spectral::Quadrature::Gauss,
+                                    Spectral::Quadrature::GaussLobatto))) {
+    CAPTURE(initial_refinement);
+    CAPTURE(quadrature);
+
+    // we create two Shells with different refinements
```
Comment (nilsvu) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/Test_Tags.cpp]: refinement -> resolution

**Review by nilsvu — CHANGES_REQUESTED** [review]
only a couple of nitpicks, squash directly

**Comment by nikwit** [comment]
thanks, I squashed it in :)

**Review by nilsvu — APPROVED** [review]

**Pull request merged**